paddle/fluid/inference/tests/api/analyzer_int8_resnet50_tester.cc

```diff
@@ -0,0 +1,106 @@
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License. */
+
+#include <fstream>
+#include <iostream>
+#include "paddle/fluid/inference/tests/api/tester_helper.h"
+
+namespace paddle {
+namespace inference {
+namespace analysis {
+
+template <typename T>
+constexpr paddle::PaddleDType GetPaddleDType();
+
+template <>
+constexpr paddle::PaddleDType GetPaddleDType<int64_t>() {
+  return paddle::PaddleDType::INT64;
+}
+
+template <>
+constexpr paddle::PaddleDType GetPaddleDType<float>() {
+  return paddle::PaddleDType::FLOAT32;
+}
+
+void SetConfig(AnalysisConfig *cfg) {
+  cfg->SetModel(FLAGS_infer_model);
+  cfg->SetProgFile("__model__");
+  cfg->DisableGpu();
+  cfg->SwitchIrOptim();
+  cfg->SwitchSpecifyInputNames();
+  cfg->SetCpuMathLibraryNumThreads(FLAGS_paddle_num_threads);
+}
+
+template <typename T>
+PaddleTensor LoadTensorFromStream(std::ifstream &file, std::vector<int> shape,
+                                  std::string name) {
+  PaddleTensor tensor;
+  tensor.name = name;
+  tensor.shape = std::move(shape);
+  tensor.dtype = GetPaddleDType<T>();
+  size_t numel = std::accumulate(begin(tensor.shape), end(tensor.shape), 1,
+                                 std::multiplies<T>());
+  tensor.data.Resize(numel * sizeof(T));
+
+  std::copy_n(std::istream_iterator<T>(file), numel,
+              static_cast<T *>(tensor.data.data()));
+
+  return tensor;
+}
+
+void SetInput(std::vector<std::vector<PaddleTensor>> *inputs) {
+  PADDLE_ENFORCE_EQ(FLAGS_test_all_data, 0, "Only have single batch of data.");
+  std::string line;
+  std::ifstream file(FLAGS_infer_data);
+
+  int batch_size = 100;
+  PaddleTensor data =
+      LoadTensorFromStream<float>(file, {batch_size, 3, 224, 224}, "data");
+
+  PaddleTensor labels =
+      LoadTensorFromStream<int64_t>(file, {batch_size, 1}, "labels");
+
+  std::vector<PaddleTensor> input_slots{{data, labels}};
+  inputs->emplace_back(input_slots);
+}
+
+TEST(Analyzer_int8_resnet50, quantization) {
+  std::vector<std::vector<PaddleTensor>> input_slots_all;
+  SetInput(&input_slots_all);
+  auto warmup_data =
+      std::make_shared<std::vector<PaddleTensor>>(input_slots_all[0]);
+
+  AnalysisConfig cfg;
+  SetConfig(&cfg);
+  cfg.EnableMKLDNN();
+
+  cfg.pass_builder()->SetPasses(
+      {"infer_clean_graph_pass", "mkldnn_placement_pass",
+       "depthwise_conv_mkldnn_pass", "conv_bn_fuse_pass",
+       "conv_eltwiseadd_bn_fuse_pass", "conv_bias_mkldnn_fuse_pass",
+       "conv_elementwise_add_mkldnn_fuse_pass", "conv_relu_mkldnn_fuse_pass",
+       "fc_fuse_pass", "is_test_pass"});
```

Comment:
why should SetPasses manually here?

Reply:
The default order of passes wasn't working

Reply:
i tried without it and it works now. I can remove this

Reply:
No, it gets wrong:
![image](https://user-images.githubusercontent.com/17185347/54833937-5f216780-4cbf-11e9-9138-84a11372d43c.png)

Reply:
> The default order of passes wasn't working

If default order of passes wasn't working, should you fix the default order? Otherwise, it is confused for users, since they should not care about the order.

Reply:
I will see what can be done about this.

Reply:
@luotao1 , @sfraczek , I have fixed the order of mkldnn passes: https://github.com/PaddlePaddle/Paddle/pull/16448
and removed use of `SetPasses()` here.

Reply:
@luotao1 , @sfraczek , sorry, I have abandoned https://github.com/PaddlePaddle/Paddle/pull/16448 and reverted the changes here, as it does not work well in all cases